wallet/how-to/manage-networks/use-multichain.md

```diff
@@ -0,0 +1,245 @@
+---
+description: Use the Multichain API.
+toc_max_heading_level: 4
+sidebar_custom_props:
+  flask_only: true
+---
+
+# Interact with multiple networks
+
+:::flaskOnly
+:::
+
+You can use the Multichain API to interact with multiple blockchain networks in MetaMask simultaneously.
+The API allows you to target specific chains as part of each method call, eliminating the need to
+detect and switch networks before executing signatures and transactions.
+
+:::note See also
+- [About the Multichain API](../../concepts/multichain-api.md)
+- [Multichain API reference](../../reference/multichain-api.md)
+:::
+
+## Prerequisites
+
+[Install MetaMask Flask.](/snaps/get-started/install-flask)
+
+## Steps
+
+### 1. Set up your project
+
+Establish a connection to MetaMask Flask and set up basic message handling using the
+[`wallet_notify`](../../reference/multichain-api.md#wallet_notify) event:
+
+```javascript
+// Initialize the connection to Flask.
+const EXTENSION_ID = "ljfoeinjpaedjfecbmggjgodbgkmjkjk"; // Flask extension ID
+const extensionPort = chrome.runtime.connect(EXTENSION_ID)
+
+// Set up message listener for events.
+extensionPort.onMessage.addListener((msg) => {
+  // Format wallet_notify events to be able to read them later.
+  if (msg.data.method === "wallet_notify") {
+    console.log("wallet_notify:", {
+      scope: msg.data.params.scope,
+      method: msg.data.params.notification.method,
+      subscription: msg.data.params.notification.params.subscription,
+      number: msg.data.params.notification.params.result.number
+    })
+    return;
+  }
+  console.log(msg.data)
+})
+```
+
+### 2. Manage multichain connections
+
+To interact with multiple networks simultaneously, you'll create and manage
+[CAIP-25](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-25.md) multichain connections
+with MetaMask.
+
+#### 2.1. Check existing connections
+
+Before creating a new connection, check if one already exists using the
+[`wallet_getSession`](../../reference/multichain-api.md#wallet_getsession) method.
+For example:
+
+```javascript
+extensionPort.postMessage({
+  type: "caip-348",
+  data: {
+    jsonrpc: "2.0",
+      method: "wallet_getSession",
+      params: {}
+  }
+});
+```
+
+If the result returns an empty `sessionScopes` parameter, then a multichain connection is not active
+and you must create a new connection.
+
+#### 2.2. Create a new connection
+
+Create a new connection using the [`wallet_createSession`](../../reference/multichain-api.md#wallet_createsession) method.
+Specify which chains and methods your dapp needs access to, using the `optionalScopes` parameter.
+For example:
+
+```javascript
+extensionPort.postMessage({
+  type: "caip-348",
+  data: {
+    jsonrpc: "2.0",
+    method: "wallet_createSession",
+    params: {
+      optionalScopes: {
+        "wallet:eip155": { // General Ethereum wallet functions
```

Comment:
@adonesky1 thoughts on wording here? Should we even advertise that `wallet:eip155` exists right now?

Reply:
🤔 I mean its technically still functionally useful for calling `wallet_addEthereumChain`... Why in your opinion shouldn't we?

Reply:
because `wallet:<ecosystem>` vs `<ecosystem>:<ecosystem_null_reference>` seems to be up in the air again

Reply:
ah hmmm fair point. Perhaps worth just keeping this out of the docs for now

Reply:
Removed